### ISSUE TITLE
Support AlienSwarm branch

### ISSUE THREAD

**Issue by LiberalMineral** [thexa4/source-pbr#45]
It doesn't have some methods like:
" SetModulationPixelShaderDynamicState_LinearColorSpace" 
"pShaderAPI->SetPixelShaderStateAmbientLightCube"
...

**Comment by Sectonid** [comment]
You can use the one from [Alien Swarm: Reactive Drop](https://github.com/ReactiveDrop/reactivedrop_public_src/blob/reactivedrop_beta/src/materialsystem/stdshaders/pbr_dx9.cpp)

**Comment by ghost** [comment]
ASW Reactive Drop has the one backported from Strata Source, so I recommend it

**Comment by kosah110** [comment]
> ASW Reactive Drop has the one backported from Strata Source, so I recommend it

hey will this work on gmod too? like drag and game_shader_dx9 and the pbr vcs?

**Comment by ghost** [comment]
> > ASW Reactive Drop has the one backported from Strata Source, so I recommend it
> 
> hey will this work on gmod too? like drag and game_shader_dx9 and the pbr vcs?

Garry's Mod and ASRD are completely different engine branches, so I don't know what will happen, you can test it tho

**Comment by kosah110** [comment]
> > > ASW Reactive Drop has the one backported from Strata Source, so I recommend it
> > 
> > 
> > hey will this work on gmod too? like drag and game_shader_dx9 and the pbr vcs?
> 
> Garry's Mod and ASRD are completely different engine branches, so I don't know what will happen, you can test it tho

Update it didnt :( do you have or know any improved pbr on the sdk 2013 so it works on gmod?

**Comment by ghost** [comment]
No, but you can ask sears2424 for information about his implementation, or you can create your own.

**Comment by kosah110** [comment]
> No, but you can ask sears2424 for information about his implementation, or you can create your own.

How about this? Map labs is based on sdk 2013 this should work? https://github.com/map-labs-source/Map-Labs/blob/3a9a4b8abc7be01d0fbe43d7f1b23bfed68ca7d3/sp/src/materialsystem/stdshaders/pbr_dx9.cpp

**Comment by ghost** [comment]
> > No, but you can ask sears2424 for information about his implementation, or you can create your own.
> 
> How about this? Map labs is based on sdk 2013 this should work? https://github.com/map-labs-source/Map-Labs/blob/3a9a4b8abc7be01d0fbe43d7f1b23bfed68ca7d3/sp/src/materialsystem/stdshaders/pbr_dx9.cpp

thexa4's implementation is identical to this, map labs uses this implementation - may cause same outcome

**Comment by kosah110** [comment]
> > > No, but you can ask sears2424 for information about his implementation, or you can create your own.
> > 
> > 
> > How about this? Map labs is based on sdk 2013 this should work? https://github.com/map-labs-source/Map-Labs/blob/3a9a4b8abc7be01d0fbe43d7f1b23bfed68ca7d3/sp/src/materialsystem/stdshaders/pbr_dx9.cpp
> 
> thexa4's implementation is identical to this, map labs uses this implementation - may cause same outcome

yes based on that code it has probably better features than thexa4, so i should use that one instead?

**Comment by ghost** [comment]
> > > > No, but you can ask sears2424 for information about his implementation, or you can create your own.
> > > 
> > > 
> > > How about this? Map labs is based on sdk 2013 this should work? https://github.com/map-labs-source/Map-Labs/blob/3a9a4b8abc7be01d0fbe43d7f1b23bfed68ca7d3/sp/src/materialsystem/stdshaders/pbr_dx9.cpp
> > 
> > 
> > thexa4's implementation is identical to this, map labs uses this implementation - may cause same outcome
> 
> yes based on that code it has probably better features than thexa4, so i should use that one instead?

I don't work for facepunch, don't ask me

**Comment by kosah110** [comment]
> > > > > No, but you can ask sears2424 for information about his implementation, or you can create your own.
> > > > 
> > > > 
> > > > How about this? Map labs is based on sdk 2013 this should work? https://github.com/map-labs-source/Map-Labs/blob/3a9a4b8abc7be01d0fbe43d7f1b23bfed68ca7d3/sp/src/materialsystem/stdshaders/pbr_dx9.cpp
> > > 
> > > 
> > > thexa4's implementation is identical to this, map labs uses this implementation - may cause same outcome
> > 
> > yes based on that code it has probably better features than thexa4, so i should use that one instead?
> 
> I don't work for facepunch, don't ask me

hey so i want to follow up on an update about this pbr, do you have steam or discord i can message you on?

**Comment by ghost** [comment]
> > > > > > No, but you can ask sears2424 for information about his implementation, or you can create your own.
> > > > > 
> > > > > 
> > > > > How about this? Map labs is based on sdk 2013 this should work? https://github.com/map-labs-source/Map-Labs/blob/3a9a4b8abc7be01d0fbe43d7f1b23bfed68ca7d3/sp/src/materialsystem/stdshaders/pbr_dx9.cpp
> > > > 
> > > > 
> > > > thexa4's implementation is identical to this, map labs uses this implementation - may cause same outcome
> > > 
> > > 
> > > yes based on that code it has probably better features than thexa4, so i should use that one instead?
> > 
> > 
> > I don't work for facepunch, don't ask me
> 
> hey so i want to follow up on an update about this pbr, do you have steam or discord i can message you on?

yeah, my dc is: `__osx__`, and my steam is DecalOverdose
